chain-network/src/grpc/convert.rs

```diff
@@ -92,6 +92,11 @@ pub(super) fn decode_peer(meta: &MetadataMap) -> Result<Peer, Error> {
     Ok(address.into())
 }
 
+pub(super) fn encode_peer(peer: &Peer, meta: &mut MetadataMap) {
+    let addr = format!("{}", peer.addr());
+    meta.insert(PEER_ADDRESS_HEADER, MetadataValue::from_str(&addr).unwrap());
```

Comment:
I would try not to have an `unwrap` here as the `addr().to_string()` may not be always ascii. Just in case, can we have a check on this?

Reply:
I expect the string representation of the IP address to be always ASCII. If it's not, the text header will not work and we'd have to change the metadata key anyway.